raptorWeb/authprofiles/views.py

```diff
@@ -0,0 +1,347 @@
+                except IndexError:
+                    instance_dict.update({
+                    "displayed_profile": {
+                        "base": {
+                            "username": user_base.username,
+                            "date_joined": user_base.date_joined,
+                            "last_login": user_base.last_login,
+                            "is_staff": user_base.is_staff
+                        },
+                        "extra": {
+                            "mc_username": user_extra.minecraft_username,
+                            "favorite_pack": user_extra.favorite_modpack
+                        }
+                    }
+                }) 
+            except User.DoesNotExist:
+                try:
+                    discord_user = DiscordUserInfo.objects.get(username=profile_name)
+                    instance_dict.update({
+                        "displayed_profile": {
+                            "base": {
+                                "username": discord_user.username,
+                                "date_joined": discord_user.date_joined,
+                                "last_login": discord_user.last_login,
+                            },
+                            "extra": {
+                                "picture": f'https://cdn.discordapp.com/avatars/{discord_user.id}/{discord_user.profile_picture}.png',
+                                "mc_username": discord_user.minecraft_username,
+                                "discord_username": discord_user.tag,
+                                "favorite_pack": discord_user.favorite_modpack
+                            }
+                        }
+                    })
+                except DiscordUserInfo.DoesNotExist:
+                    return HttpResponse("A User with the provided username was not found")
+
+            return render(request, self.template_name, context=instance_dict)
+
+        else: 
+            return redirect('/accessdenied')
+
+    def post(self, request, profile_name):
+
+        profile_edit_form = DiscordUserInfoForm(request.POST)
+        extra_edit_form = UserProfileInfoForm(request.POST)
+        instance_dict = player_poller.currentPlayers_DB
+        instance_dict["profile_edit_form"] = profile_edit_form
+        instance_dict["extra_edit_form"] = self.extra_edit_form
+        instance_dict = viewContext.update_context(context = instance_dict)
+        if profile_edit_form.is_valid() and extra_edit_form.is_valid():
+            LOGGER.info("A User modified their profile details")
+            changed_user = None
+            try:
+                changed_user = DiscordUserInfo.objects.get(tag=request.user)
+            except ObjectDoesNotExist:
+                changed_user_base = User.objects.get(username=request.user)
+                changed_user = UserProfileInfo.objects.get(user=changed_user_base)
+            if profile_edit_form.cleaned_data["minecraft_username"] != '':
```

Comment:
Looping over all form fields in a list of some type and checking for changes rather than individual `if` blocks for each field may be a better solution here